lib/board.rb

```diff
@@ -17,4 +17,34 @@ def cell_generator(rows, columns)
     end
     @cells = generated_cells
   end
+
+  def valid_coordinate?(coordinate)
+    @cells.keys.include?(coordinate)
+  end
+
+  def valid_placement?(ship, coordinates)
```

Comment:
we should also add in valid coordinate into the valid placement method and test for that.

Reply:
Added!